# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../mine/gasgui-use-dialog-config.patch
# Copyright (C) 2011 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

Description: Use dialog-config to determine the needed LDFLAGS for libdialog

This way we do not have to worry about if we have to link against
the narrowc or the widec version of libncurses.

--- ./Makefile.orig	2011-03-23 17:33:23.964072236 +0100
+++ ./Makefile	2011-03-23 18:48:53.956072238 +0100
@@ -22,6 +22,7 @@
 #
 GAS_VER = $(MINE_VER)
 GAS_OBJ = gasgui.o readdb.o selector.o memdb.o
+GAS_DIALOG_LDFLAGS = $(shell dialog-config --libs)
 
 # Let's use AVL for the memory DB
 #
@@ -83,7 +84,7 @@
 	$(CC) -static $(MINE_ALL_OBJS) $(MINE_ALL_LIBS) -o mine.static
 
 gasgui: $(GAS_OBJ)
-	$(CC) $(GAS_OBJ) -ldialog -lcurses -lm -o gasgui
+	$(CC) $(GAS_OBJ) $(GAS_DIALOG_LDFLAGS) -o gasgui
 
 install: all
 	install mine $(sysprefix)/bin/mine
